src/lib/metricsMgr/MetricsManager.cpp

```diff
@@ -131,15 +134,54 @@ void MetricsManager::add(const std::string& srv, const std::string& subServ, con
     return;
   }
 
+  // FIXME P4: See github issue #2781
+  if (tenantCheck(srv) != "OK")
+  {
+    return;
+  }
+
+  // FIXME P4: See github issue #2781
+  ConnectionInfo ci;
+  ci.httpHeaders.servicePathReceived = true;
+  if (servicePathCheck(&ci, subServ.c_str()) != 0)
+  {
+    return;
+  }
+
   //
   // Exclude the first '/' from the Sub Service
   // But, only if it starts with a '/'
   //
-  const char* subService = subServ.c_str();
+  // Also, if service path ends in '/#', cancel out that part
+  //
+  // Note that for the 'cancel out' part we need to allocate a copy of
+  // the service path (and frre it after usage).
+  //
+  char* subService;
+
+  if (subServ.c_str()[0] == '/')
+  {
+    subService = strdup(&subServ.c_str()[1]);
+  }
+  else
+  {
+    subService = strdup(subServ.c_str());
+  }
+
+  //
+  // Now, if service path ends in '/#', cancel out that part
+  //
+  int subServiceLen = strlen(subService);
 
-  if (subService[0] == '/')
+  if ((subServiceLen >= 2) && (subService[subServiceLen - 1] == '#') && (subService[subServiceLen - 2] == '/'))
```

Comment:
Side-comment, but meanifull for this PR anyway: 

What about "/A#"? They are previously detected as invalid service paths at the very beginning (just after reading it from the header)? What documentation says about the correct syntax?